src/V3LinkCells.cpp

```diff
@@ -292,10 +336,76 @@ class LinkCellsVisitor final : public VNVisitor {
         }
     }
 
+    AstCell* findCellByHier(const string& hierPath) {
+        std::stringstream ss(hierPath);
+        string top;
+        bool topFound = false;
+        std::getline(ss, top, '.');
+        for (auto const& pair : m_state.m_designs) {
+            if (top == pair.second) {
+                topFound = true;
+                break;
+            }
+        }
+        if (!topFound) {
+            v3error("Can't find top-level module for instance path: '" << hierPath << "'");
```

Comment:
Pass nodep or a FileLine down so user can see what source caused this.

```suggestion
            nodep->v3error("Can't find top-level module for instance path: '" << hierPath << "'");
```